include/kodlab_mjbots_sdk/lcm_subscriber.h

```diff
@@ -1,85 +1,102 @@
-// BSD 3-Clause License
-// Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
-// Authors:
-// Shane Rozen-Levy <srozen01@seas.upenn.edu>
+/**
+ * @file lcm_subscriber.h
+ * @author Ethan Musser (emusser@seas.upenn.edu)
+ * @author J. Diego Caporale (jdcap@seas.upenn.edu)
+ * @author Shane Rozen-Levy (srozen01@seas.upenn.edu)
+ * @brief Provides LCM subscriber object for subscribing to onw or multiple LCM
+ * channels.
+ * @date 7/7/22
+ *
+ * @copyright Copyright 2022 The Trustees of the University of Pennsylvania. All
+ * rights reserved. BSD 3-Clause License
+ *
+ */
 
 #pragma once
+
+#include <map>
+#include <mutex>
 #include <string>
 #include "lcm/lcm-cpp.hpp"
 #include "real_time_tools/thread.hpp"
 #include "kodlab_mjbots_sdk/abstract_realtime_object.h"
+#include "kodlab_mjbots_sdk/lcm_message_handler.h"
+
+namespace kodlab
+{
 
-namespace kodlab {
-/*!
- * @brief An template class for an lcm subscriber. Subscribes to just one msg of type msg_type.
- * @tparam MessageClass the lcm msg type
+/**
+ * @brief LCM subscriber capable of subscribing to multiple channels
+ * @todo Implement subscription removal (i.e., unsubscribing)
  */
-template<class MessageClass>
-class LcmSubscriber : public AbstractRealtimeObject {
- public:
+class LcmSubscriber : public AbstractRealtimeObject
```

Comment:
since this class is no longer templated, add a .cpp and keep the implementation in the .cpp.

Reply:
Split in a0b1d74091ed316d4b7e7abb61070ece3161ebc9.